packages/api/tests/endpoints/test-executions.js

```diff
@@ -170,3 +369,240 @@ test('GET fails if execution is not found', async (t) => {
   t.is(response.status, 404);
   t.true(response.body.message.includes('No record found for'));
 });
+
+test('DELETE removes only specified execution from all data stores', async (t) => {
+  const { knex } = t.context;
+
+  const newExecution = fakeExecutionFactoryV2({
+    arn: 'arn3',
+    status: 'completed',
+    name: 'test_execution',
+  });
+
+  fakeExecutions.push(newExecution);
+
+  await Promise.all(
+    fakeExecutions.map(async (execution) => {
+      // delete async operation foreign key to avoid needing a valid async operation
+      delete execution.asyncOperationId;
+      await executionModel.create(execution);
+      const executionPgRecord = await translateApiExecutionToPostgresExecution(
+        execution,
+        knex
+      );
+      await executionPgModel.create(knex, executionPgRecord);
+    })
+  );
+
+  t.true(await executionModel.exists({ arn: newExecution.arn }));
+  t.true(await executionPgModel.exists(knex, { arn: newExecution.arn }));
+
+  await request(app)
+    .delete(`/executions/${newExecution.arn}`)
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${jwtAuthToken}`)
+    .expect(200);
+
+  // Correct Dynamo and PG execution was deleted
+  t.false(await executionModel.exists({ arn: newExecution.arn }));
+
+  const dbRecords = await executionPgModel.search(t.context.knex, {
+    arn: newExecution.arn,
+  });
+
+  t.is(dbRecords.length, 0);
+
+  // Previously created executions still exist
+  t.true(await executionModel.exists({ arn: fakeExecutions[0].arn }));
+  t.true(await executionModel.exists({ arn: fakeExecutions[1].arn }));
+
+  const originalExecution1 = await executionPgModel.search(t.context.knex, {
+    arn: fakeExecutions[0].arn,
+  });
+
+  t.is(originalExecution1.length, 1);
+
+  const originalExecution2 = await executionPgModel.search(t.context.knex, {
+    arn: fakeExecutions[1].arn,
+  });
+
+  t.is(originalExecution2.length, 1);
+});
+
+test('DELETE returns a 404 if Dynamo execution cannot be found', async (t) => {
+  const nonExistantExecution = {
+    arn: 'arn9',
+    status: 'completed',
+    name: 'test_execution',
+  };
+
+  const response = await request(app)
+    .delete(`/executions/${nonExistantExecution.arn}`)
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${jwtAuthToken}`)
+    .expect(404);
+
+  t.is(response.body.message, 'No record found');
+});
+
+test('GET /history/:granuleId returns all workflow names associated with the granule', async (t) => {
+  const { fakeGranules, fakePGExecutions } = t.context;
+
+  const response = await request(app)
+    .get(`/executions/history/${fakeGranules[0].granuleId}`)
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${jwtAuthToken}`);
+
+  response.body.forEach((execution, index) => {
+    t.is(execution.arn, fakePGExecutions[index].arn);
+    t.is(execution.workflow_name, fakePGExecutions[index].workflow_name);
```

Comment:
Given that we now are returning full executions from the API, is there any reason not to do this?

```suggestion
    t.deepEqual(execution, fakePGExecutions[index]);
```

This comment applies throughout the tests

Reply:
Was having issues with the date fields not being the same type, but I did a refactor to consolidate some of the fakeExecutions v. fakePGExecutions and also filtered out those fields for the comparison, so this should be solved now for all the tests